DEVELOP.md

```diff
@@ -0,0 +1,485 @@
+should be covered by unit tests or TDML tests.  Daffodil naming
+conventions and more are covered in the Confluence pages [Code Style
+Guidelines](https://cwiki.apache.org/confluence/display/DAFFODIL/Code+Style+Guidelines)
+and [Coding for
+Performance](https://cwiki.apache.org/confluence/display/DAFFODIL/Coding+for+Performance).
+
+### Code Workflow
+
+It is very important that you fork the Apache repository to your own
+GitHub account before you start making changes to Apache Daffodil
+source code.  You should always make changes only to your own fork of
+the upstream Apache repository, that is, clone your working copy from
+your fork and push commits back to your own fork only.  The same
+workflow applies to Apache Daffodil's other two source code
+repositories, that is, always fork these repositories and clone from
+your forks before you make any changes to the [Daffodil VS Code
+Extension](https://github.com/apache/daffodil-vscode) or [Daffodil
+Website](https://github.com/apache/daffodil-site).  When you are ready
+to contribute code from any of your forks to any of these upstream
+Apache repositories, you must create a pull request from your fork,
+allow the Apache Daffodil developers to review your contribution, make
+any suggested changes, and wait for their approval before you or
+someone else can merge your code into one of Apache's source code
+repositories.
+
+The Daffodil developers want each pull request to add only one commit
+to the Apache repository to keep its git history as readable and
+bisectable as possible.  Your pull request should fix a JIRA issue in
+its entirety but it should not fix 2 or more unrelated JIRA issues in
+the same commit either.  Your pull request should start with only one
+commit in it before you ask the developers to review it and your pull
+request should end with only one commit in it after you receive enough
+reviewer approvals (two +1's for Daffodil and Daffodil VS Code
+Extension, one +1 for Daffodil Website).  If you have made multiple
+commits to your pull request due to reviewer suggestions (which is the
+usual case), you must not merge these multiple commits into the Apache
+repository.  You must rebase and squash your pull request back to one
+commit as shown in the [Code Contributor
+Workflow](https://cwiki.apache.org/confluence/display/DAFFODIL/Code+Contributor+Workflow),
+which you should read and follow carefully to ensure a successful code
+contribution to Apache Daffodil.
+
+Before you change Daffodil code, you will need to know how to use JIRA
+as well as GitHub.  You will assign a JIRA issue to yourself before
+you start working on it to avoid redundant effort in case another
+developer decides to work on the same issue.  Then you will mark the
+issue as fixed after your pull request is merged.  Here are the steps
+you need to perform in JIRA to mark the issue resolved after your pull
+request is merged (please don't forget these steps):
```

Comment:
Usually the person who merges the PR (i.e. an ASF comitter) marks it as resolved. I'm not even sure if non-comitters can resolve commits?